Library/Homebrew/extend/os/mac/formula_cellar_checks.rb

```diff
@@ -6,7 +6,7 @@ def check_shadowed_headers
       formula.name.start_with?(formula_name)
     end
 
-    return if formula.name =~ /^php\d+$/
+    return if formula.name =~ /^php(?:\@\d\.\d|\d+)?$/
```

Comment:
> So the at sign doesn't need to be escaped ?

Yup.

> We could do /^php(\d+|@\d+.\d+)?$/ which would also match 7.10 if the version would be written with double digits.

That works. Could also do `/^php@?\d+(.\d+)?$/` if you think that's more readable (I do but you may disagree. Either is fine, though 👍